yt_dlp/extractor/bbc.py

```diff
@@ -438,41 +438,76 @@ def _process_media_selector(self, media_selection, programme_id):
+            for tag, subformats in (version_subtitles or {}).items():
+                subtitles.setdefault(tag, []).extend(subformats)
+            return formats, subtitles
 
+        # Fallback when there is no VPID in json response.
+        if not traverse_obj(playlist, ('defaultAvailableVersion')):
+            episode_block = self._search_regex(
+                r'"episode"\s*:\s*\{(.*?)\}\s*,\s*"relatedEpisodes"',
+                webpage,
+                'episode',
+                fatal=False,
+            )
+            episode_json = {}
+            if episode_block:
+                episode_json = self._parse_json('{' + episode_block + '}', playlist_id)
+
+            description = traverse_obj(episode_json, ('synopses', 'large'))
+            programme_id = self._search_regex(
+                r'"id"\s*:\s*"([0-9a-z]{8})"',
+                self._search_regex(
+                    r'"versions"\s*:\s*\[\s*(\{[^]]*"id"\s*:\s*"[0-9a-z]{8}".*?)\]\s*,',
+                    webpage,
+                    'versions',
+                    fatal=False,
+                ),
+                'pid',
+                fatal=False)
+            duration = int_or_none(self._search_regex(
+                r'<span[^>]*episode-metadata__text[^>]*>\s*([^<]+)\s*</span>',
+                webpage,
+                'duration',
+                fatal=False,
+            ))
+
+            formats, subtitles = _parse_formats(programme_id)
+        else:
             for version in playlist.get('allAvailableVersions', []):
                 smp_config = version['smpConfig']
-                title = smp_config['title']
+                title = smp_config['title'] or None
                 description = smp_config['summary']
                 for item in smp_config['items']:
                     kind = item['kind']
                     if kind not in ('programme', 'radioProgramme'):
                         continue
                     programme_id = item.get('vpid')
                     duration = int_or_none(item.get('duration'))
-                    version_formats, version_subtitles = self._download_media_selector(programme_id)
                     types = version['types']
-                    for f in version_formats:
-                        f['format_note'] = ', '.join(types)
-                        if any('AudioDescribed' in x for x in types):
-                            f['language_preference'] = -10
-                    formats += version_formats
-                    for tag, subformats in (version_subtitles or {}).items():
-                        subtitles.setdefault(tag, []).extend(subformats)
-
-            return programme_id, title, description, duration, formats, subtitles
-        except ExtractorError as ee:
-            if not (isinstance(ee.cause, HTTPError) and ee.cause.status == 404):
-                raise
-
-        # fallback to legacy playlist
-        return self._process_legacy_playlist(playlist_id)
```

Comment:
why are we removing the fallback?

`_process_legacy_playlist` is now dead code